conf/machine/cubox-i.conf

```diff
@@ -26,7 +26,7 @@ SPL_BINARY = "SPL"
 WKS_FILES = "imx-uboot-spl.wks.in"
 
 UBOOT_EXTLINUX = "1"
-UBOOT_EXTLINUX_ROOT = "root=PARTUUID=${uuid}"
```

Comment:
The partuuid is failing? It should work.

Reply:
yes, with partuuid hummingboard and cubox-i don't boot due to following kernel panic:

```
[    4.753429] b300        30183936 mmcblk1 
[    4.753434]  driver: mmcblk
[    4.760270]   b301          139575 mmcblk1p1 e662228b-01
[    4.760273] 
[    4.767138] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[    4.775644] CPU0: stopping
[    4.778461] CPU: 0 PID: 0 Comm: swapper/0 Not tainted 5.4.66+g442e6020fa8a #1
[    4.785610] Hardware name: Freescale i.MX6 Quad/DualLite (Device Tree)
[    4.792171] [<c0112508>] (unwind_backtrace) from [<c010cac8>] (show_stack+0x10/0x14)
[    4.799933] [<c010cac8>] (show_stack) from [<c0c87bb0>] (dump_stack+0xe0/0x114)
[    4.807260] [<c0c87bb0>] (dump_stack) from [<c0110ab0>] (handle_IPI+0x348/0x3b0)
[    4.814677] [<c0110ab0>] (handle_IPI) from [<c0546780>] (gic_handle_irq+0x8c/0x9c)
[    4.822265] [<c0546780>] (gic_handle_irq) from [<c0101a70>] (__irq_svc+0x70/0x98)
[    4.829760] Exception stack(0xc1301ed8 to 0xc1301f20)
[    4.834825] 1ec0:                                                       00000001 00000006
[    4.843017] 1ee0: 00000000 c130c780 00000001 c130fb34 1ca4a466 ef7bcdb8 c13eb288 00000001
[    4.851209] 1f00: 00000001 1c79ccf4 00000000 c1301f28 c0184a68 c08c9cd8 20000013 ffffffff
[    4.859406] [<c0101a70>] (__irq_svc) from [<c08c9cd8>] (cpuidle_enter_state+0x16c/0x50c)
[    4.867513] [<c08c9cd8>] (cpuidle_enter_state) from [<c08ca0bc>] (cpuidle_enter+0x30/0x40)
[    4.875794] [<c08ca0bc>] (cpuidle_enter) from [<c0161f00>] (do_idle+0x1ec/0x2ec)
[    4.883204] [<c0161f00>] (do_idle) from [<c01623b4>] (cpu_startup_entry+0x18/0x1c)
[    4.890793] [<c01623b4>] (cpu_startup_entry) from [<c1200e14>] (start_kernel+0x404/0x4dc)
[    4.898985] [<c1200e14>] (start_kernel) from [<00000000>] (0x0)
[    4.904936] ---[ end Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0) ]---
```

Reply:
Good but it all depends if the kernel has the `uuid` support and if the U-Boot in use is adding it. Use a fixed device is not good as it is fragile and easy to fail. 

What U-Boot are you basing one? 

The `u-boot-fslc` provides support for it, as you can see at https://github.com/Freescale/u-boot-fslc/blob/2020.04%2Bfslc/include/configs/mx6cuboxi.h#L73 and https://github.com/Freescale/u-boot-fslc/blob/2020.04%2Bfslc/configs/mx6cuboxi_defconfig#L23

So it seems something is weird in your side.

Reply:
I am doing a straight-forward build of `core-image-base` with default configurations. Here is what I append to `local.conf`:
```
MACHINE = "cubox-i"

DISTRO_FEATURES_append = " systemd"
VIRTUAL-RUNTIME_init_manager = "systemd"
DISTRO_FEATURES_BACKFILL_CONSIDERED = "sysvinit"
VIRTUAL-RUNTIME_initscripts = ""

ACCEPT_FSL_EULA = "1"
LICENSE_FLAGS_WHITELIST = "commercial"

IMAGE_INSTALL_append = "alsa-utils alsa-utils-speakertest"
```

Am I missing something?

Reply:
It seems to be right. Are you using the generated wic file?

You need to check the boot sequence of the boot to find out if it is failing for any reason. Running the `finduuid` on U-Boot prompt may help.